File: org.kevoree.modeling.optimization.framework.core/src/main/java/org/kevoree/modeling/optimization/api/fitness/FitnessFunction.java
package org.kevoree.modeling.optimization.api.fitness;

import org.kevoree.modeling.api.KMFContainer;
import org.kevoree.modeling.optimization.api.GenerationContext;

/**
 * Created by duke on 4/7/14.
 */
public interface FitnessFunction<A extends KMFContainer> {

    public double evaluate(A model, GenerationContext<A> context);

}

File: org.kevoree.modeling.optimization.framework.core/src/main/java/org/kevoree/modeling/optimization/api/fitness/FitnessFunction.kt
package org.kevoree.modeling.optimization.api.fitness;

import org.kevoree.modeling.api.KMFContainer
import org.kevoree.modeling.optimization.api.GenerationContext

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 12/02/13
 * Time: 14:40
 */
public trait FitnessFunction<A : KMFContainer> {

    public fun evaluate(model: A, context: GenerationContext<A>): Double ;

}
